projects/aca-folder-rules/src/lib/services/folder-rules.service.spec.ts

```diff
@@ -0,0 +1,185 @@
+            enabled: true
+          }
+        },
+        {
+          entry: {
+            shared: false,
+            cascade: false,
+            asynchronous: false,
+            name: 'rule2',
+            id: 'e0e645ca-e6c0-47d4-9936-1a8872a6c30b',
+            triggers: ['INBOUND'],
+            actions: [
+              {
+                actionDefinitionId: 'move',
+                params: {
+                  'destination-folder': '279c65f0-912b-4563-affb-ed9dab8338e0',
+                  actionContext: 'rule'
+                }
+              }
+            ],
+            enabled: true
+          }
+        }
+      ]
+    }
+  };
+  const dummyRules: Partial<Rule>[] = [
+    {
+      id: 'd388ed54-a522-410f-a158-6dbf5a833731',
+      name: 'rule1',
+      description: '',
+      enabled: true,
+      cascade: false,
+      asynchronous: false,
+      errorScript: '',
+      shared: false,
+      triggers: ['INBOUND'],
+      conditions: null,
+      actions: [
+        {
+          actionDefinitionId: 'copy',
+          params: {
+            'deep-copy': false,
+            'destination-folder': '279c65f0-912b-4563-affb-ed9dab8338e0',
+            actionContext: 'rule'
+          }
+        }
+      ]
+    },
+    {
+      id: 'e0e645ca-e6c0-47d4-9936-1a8872a6c30b',
+      name: 'rule2',
+      description: '',
+      enabled: true,
+      cascade: false,
+      asynchronous: false,
+      errorScript: '',
+      shared: false,
+      triggers: ['INBOUND'],
+      conditions: null,
+      actions: [
+        {
+          actionDefinitionId: 'move',
+          params: {
+            'destination-folder': '279c65f0-912b-4563-affb-ed9dab8338e0',
+            actionContext: 'rule'
+          }
+        }
+      ]
+    }
+  ];
+
+  const firstRuleId = dummyResponse.list.entries[0].entry.id;
+  const lastRuleId = dummyResponse.list.entries[dummyResponse.list.entries.length - 1].entry.id;
+  const nodeId = '';
+  const ruleSetId = '-default-';
+
+  beforeEach(async () => {
+    TestBed.configureTestingModule({
+      imports: [CoreTestingModule],
+      providers: [FolderRulesService]
+    });
+
+    folderRulesService = TestBed.inject<FolderRulesService>(FolderRulesService);
+
+    apiCallSpy = spyOn(folderRulesService, 'apiCall').and.returnValue(of(dummyResponse) as any);
+
+    rulesPromise = folderRulesService.rulesListing$.pipe(take(2)).toPromise();
+
+    folderRulesService.loadAllRules(nodeId, ruleSetId);
+
+    rules = await rulesPromise;
+  });
+
+  it('should set data', async () => {
+    expect(rules).toBeTruthy('rulesListing$ is empty');
+    expect(rules.length).toBe(2, 'rulesListing$ size is wrong');
+  });
+
+  it('should make 1 API call', async () => {
+    expect(apiCallSpy).toHaveBeenCalledTimes(1);
+  });
+
+  it('should format response to list of rules', async () => {
+    expect(rules).toEqual(dummyRules, 'The list of rules is incorrectly formatted');
+  });
+
+  it('should format each rule entity', async () => {
```

Comment:
This test is unnecessary as it's already covered by the previous one. If `rules` is different to `dummyRules` in any way, including nested objects, the statement on line 171 should fail.